js/utils.js

```diff
@@ -1025,6 +1025,170 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
+    }
+
+    var records = _.clone(options.records);
+    var isSortAsc = options.sortOrder === 'asc';
+    var sortField = options.sortField;
+    var startsWithAlphabet = /^[A-Z,a-z]/;
+    var sortType = getFieldType(records[0][sortField]);
+  
+    return records.sort(function(a, b) {
+      var aValue = a.data[sortField];
+      var bValue = b.data[sortField];
+      
+      switch (sortType) {
+        case 'string': 
+          /**
+           * By adding '{' to the start of the string we are pushing that string to the end the sort order,
+           * and when we are adding '}' we push it even further to the end
+          */
+          aValue = aValue ? aValue.toUpperCase() : '}';
+          bValue = bValue ? bValue.toUpperCase() : '}';
+
+          if (!startsWithAlphabet.test(bValue)) {
+            bValue = '{' + bValue;
+          }
+
+          if (!startsWithAlphabet.test(aValue)) {
+            aValue = '{' + aValue;
+          }
+          
+          if (aValue < bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+          if (aValue > bValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          return 0;
+        case 'number':
+          if (!parseInt(aValue, 10)) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!parseInt(bValue, 10)) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+        case 'time':
+          if (!aValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (aValue < bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (aValue > bValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          return 0;
+        case 'date':
+          aValue = moment(aValue).valueOf();
+          bValue = moment(bValue).valueOf();
+
+          if (!aValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+         default: 
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+      }
+    });
+  }
+
+  function sortRecordsByField(options) {
+    var sortedRecords = sortByField(options);
+
+    sortHTMLElements({
+      $layoutContainer: options.$container,
+      listContainer: options.listContainer,
+      sortedRecords: sortedRecords
+    });
+  }
+
+  function sortHTMLElements(options) {
+    var sortedItemsList = [];
+    var $list = options.$layoutContainer.find(options.listContainer).detach();
+    var $listItems = $list.children();
+
+    $listItems.each(function() {
+      var $listItem = $(this);
+      var itemId = $listItem.data('entry-id');
```

Comment:
Skip the jQuery object wrapping so you can do this. We're looking at a performance-sensitive operation here so it's best to cut down on unnecessary operations.

```suggestion
      var itemId = parseInt(this.dataset.entryId, 10);
```